fe/fe-core/src/test/java/com/starrocks/planner/MaterializedViewManyJoinTest.java

```diff
@@ -0,0 +1,199 @@
+                        " p1_col4 decimal128(22, 2) NULL COMMENT \"\"\n" +
+                        ") ENGINE=OLAP \n" +
+                        "DUPLICATE KEY(dt, p1_col1)\n" +
+                        "PARTITION BY RANGE(dt)\n" +
+                        "(PARTITION p20221230 VALUES [(\"2022-12-30\"), (\"2022-12-31\")),\n" +
+                        "PARTITION p20230331 VALUES [(\"2023-03-31\"), (\"2023-04-01\")))\n" +
+                        "DISTRIBUTED BY HASH(dt, p1_col2) BUCKETS 1 \n" +
+                        "PROPERTIES (\n" +
+                        "\"replication_num\"=\"1\"" +
+                        ");");
+
+        for (int i = 1; i < 20; i++) {
+            starRocksAssert.withTable(
+                    MessageFormat.format(
+                            "CREATE TABLE tbl_{0} (\n" +
+                                    " dt date NULL COMMENT \"\",\n" +
+                                    " p{0}_col1 varchar(240) NULL COMMENT \"\",\n" +
+                                    " p{0}_col2 varchar(240) NULL COMMENT \"\"\n" +
+                                    ") ENGINE=OLAP \n" +
+                                    "DUPLICATE KEY(dt, p{0}_col1)\n" +
+                                    "PARTITION BY RANGE(dt)\n" +
+                                    "   (PARTITION p202212 VALUES [(\"2022-12-01\"), (\"2023-01-01\")),\n" +
+                                    "       PARTITION p202301 VALUES [(\"2023-01-01\"), (\"2023-02-01\")),\n" +
+                                    "       PARTITION p202302 VALUES [(\"2023-02-01\"), (\"2023-03-01\")))\n" +
+                                    "DISTRIBUTED BY HASH(dt, p{0}_col2) BUCKETS 1 \n" +
+                                    "PROPERTIES (\n" +
+                                    "\"replication_num\"=\"1\",\n" +
+                                    "\"in_memory\"=\"false\",\n" +
+                                    "\"storage_format\"=\"DEFAULT\",\n" +
+                                    "\"enable_persistent_index\"=\"false\",\n" +
+                                    "\"compression\"=\"LZ4\"\n" +
+                                    ")", i));
+        }
+    }
+
+    @ParameterizedTest(name = "{index}-{0}")
+    @MethodSource("generateManyJoinArguments")
+    @Timeout(5)
+    public void testManyJoins(String name, String mvQuery, String query, boolean expectHitMv) throws Exception {
+        LOG.info("create mv {}", mvQuery);
+        String mvName = "mv_manyjoin";
+        String createMv = "CREATE MATERIALIZED VIEW " + mvName + "\n" +
+                "REFRESH  DEFERRED MANUAL \n" +
+                "PROPERTIES (\n" +
+                "\"replication_num\"=\"1\"\n" +
+                ")\n" +
+                "AS " + mvQuery;
+        starRocksAssert.withMaterializedView(createMv);
+
+        Stopwatch watch = Stopwatch.createStarted();
+        // Make sure it's not empty
+        starRocksAssert.query(query).explainContains("OlapScanNode");
```

Comment:
Is it that possible MV is matched with the query? If not, why?

Reply:
it's possible, but since I just want to test the planner performance, I don't want to check the query plan or rewritten plan.